cli/generate.php

```diff
@@ -77,20 +83,89 @@
 
 $recipefile = $options['recipe'];
 
-// Expanding '~' on Unix-like OS'es.
-if ($recipefile[0] === '~') {
-    $homedir = getenv('HOME');
-    $recipefile = $homedir.substr($recipefile, 1);
-}
-
-$recipefile = realpath($recipefile);
+$recipefile = expand_path($recipefile);
 if ($recipefile === false) {
     echo("\nInvalid recipe file!\n");
     echo $help;
     die();
 }
 
-$loglevels = Logger::getLevels();
+if (!is_readable($recipefile)) {
+    echo("\nRecipe file not readable!\n");
```

Comment:
I just noticed all the `echo`s here. In CLI scripts, it is preferred to use `cli_write()`, `cli_writeln()` and other outputting functions from the clilib.